language/stdlib/modules/AccountLimits.move

```diff
@@ -79,89 +96,72 @@ module AccountLimits {
         time_period: u64
     ) {
         move_to(
             account,
-            LimitsDefinition {
-                max_total_flow,
+            LimitsDefinition<CoinType> {
+                max_inflow,
+                max_outflow,
                 max_holding,
                 time_period,
-                is_certified: false,
             }
         )
     }
 
     // Unrestricted accounts are represented by setting all fields in the
     // limits definition to u64 max.
-    public fun publish_unrestricted_limits(account: &signer) {
-        let u64_max = 18446744073709551615u64;
-        publish_limits_definition(account, u64_max, u64_max, u64_max)
-    }
-
-    // Removes the limits definition at the sender's address.
-    public fun unpublish_limits_definition(account: &signer)
-    acquires LimitsDefinition {
-        LimitsDefinition {
-            max_total_flow: _,
-            max_holding: _,
-            time_period: _,
-            is_certified: _,
-        } = move_from<LimitsDefinition>(Signer::address_of(account));
+    public fun publish_unrestricted_limits<CoinType>(
+        account: &signer,
+        cap: &CallingCapability,
+    ) {
+        publish_limits_definition<CoinType>(account, cap, U64_MAX, U64_MAX, U64_MAX, TIME_WINDOW)
     }
 
-    public fun update_limits_definition(
+    public fun update_limits_definition<CoinType>(
         _: &Capability<TreasuryComplianceRole>,
-        new_max_total_flow: u64,
+        limit_address: address,
+        new_max_inflow: u64,
+        new_max_outflow: u64,
         new_max_holding_balance: u64,
     ) acquires LimitsDefinition {
         // As we don't have Optionals for txn scripts, in update_unhosted_wallet_limits.move
         // we use 0 value to represent a None (ie no update to that variable)
-        if (new_max_total_flow != 0) {
-            borrow_global_mut<LimitsDefinition>(CoreAddresses::TREASURY_COMPLIANCE_ADDRESS()).max_total_flow = new_max_total_flow;
-        };
-        if (new_max_holding_balance != 0) {
-            borrow_global_mut<LimitsDefinition>(CoreAddresses::TREASURY_COMPLIANCE_ADDRESS()).max_holding = new_max_holding_balance;
-        };
-    }
-
-    // Certify the limits definition published under the account at
-    // `limits_addr`.
-    public fun certify_limits_definition(_: &Capability<TreasuryComplianceRole>, limits_addr: address)
-    acquires LimitsDefinition {
-        borrow_global_mut<LimitsDefinition>(limits_addr).is_certified = true;
-    }
-
-    // Decertify the limits_definition published under the account at
-    // `limits_addr`.
-    public fun decertify_limits_definition(_: &Capability<TreasuryComplianceRole>, limits_addr: address)
-    acquires LimitsDefinition {
-        borrow_global_mut<LimitsDefinition>(limits_addr).is_certified = false;
+        let limits_def = borrow_global_mut<LimitsDefinition<CoinType>>(limit_address);
+        if (new_max_inflow > 0) limits_def.max_inflow = new_max_inflow;
```

Comment:
Nit: can we use curly braces when the conditional is used as an imperative statement? I thought our style guidelines omitting the curly braces when conditonals were used as expressions, but not otherwise.

Reply:
good point! I'll update.